Common/TableProducer/PID/pidTOFFullRun3.cxx

```diff
@@ -0,0 +1,339 @@
+    histos.add("goodforevtime/expP", "expP", kTH2F, {pAxis, pAxis});
+    histos.add("goodforevtime/mass", "mass", kTH1F, {massAxis});
+    histos.add("goodforevtime/tofSignalPerCollision", "tofSignalPerCollision", kTH2S, {collisionAxis, tofSignalAxis});
+  }
+
+  using Trks = soa::Join<aod::Tracks, aod::TracksExtra, aod::TracksCov, aod::TOFSignal, aod::TOFEvTime, aod::TrackSelection>;
+  int ncolls = 0;
+  void process(aod::Collision const&, Trks const& tracks)
+  {
+    histos.fill(HIST("eventSelection"), 0.5f);
+    bool eventSet = false;
+    for (auto& t : tracks) {
+      if (!t.isGlobalTrack()) {
+        continue;
+      }
+      histos.fill(HIST("tracks/p"), t.p());
+      histos.fill(HIST("tracks/pt"), t.pt());
+      histos.fill(HIST("tracks/length"), t.length());
+      if (!t.hasTOF()) {
+        continue;
+      }
+      const float beta = o2::pid::tof::Beta<Trks::iterator>::GetBeta(t, t.tofEvTime());
+      const float mass = o2::pid::tof::TOFMass<Trks::iterator>::GetTOFMass(t.p(), beta);
+      histos.fill(HIST("withtof/p"), t.p());
+      histos.fill(HIST("withtof/pt"), t.pt());
+      histos.fill(HIST("withtof/length"), t.length());
+      histos.fill(HIST("withtof/tofSignal"), t.tofSignal());
+      histos.fill(HIST("withtof/beta"), t.p(), beta);
+      histos.fill(HIST("withtof/delta"), t.p(), t.tofSignal() - t.tofEvTime() - o2::pid::tof::ExpTimes<Trks::iterator, PID::Pion, false>::GetExpectedSignal(t));
+      histos.fill(HIST("withtof/expP"), t.p(), t.tofExpMom());
+      histos.fill(HIST("withtof/mass"), mass);
+      histos.fill(HIST("withtof/tofSignalPerCollision"), ncolls % 6000, t.tofSignal());
+      if (!eventSet) {
+        histos.fill(HIST("eventTime"), t.tofEvTime());
+        histos.fill(HIST("eventTimeReso"), t.tofEvTimeErr());
+        histos.fill(HIST("eventTimeMult"), t.tofEvTimeMult());
+        histos.fill(HIST("collisionTime"), t.collision().collisionTime());
+        histos.fill(HIST("collisionTimeRes"), t.collision().collisionTimeRes());
+        eventSet = true;
+        ncolls++;
+      }
+      if (!filterForTOFEventTime(t)) {
+        continue;
+      }
+      histos.fill(HIST("goodforevtime/p"), t.p());
+      histos.fill(HIST("goodforevtime/pt"), t.pt());
+      histos.fill(HIST("goodforevtime/length"), t.length());
+      histos.fill(HIST("goodforevtime/tofSignal"), t.tofSignal());
+      histos.fill(HIST("goodforevtime/beta"), t.p(), beta);
+      histos.fill(HIST("goodforevtime/delta"), t.p(), t.tofSignal() - t.tofEvTime() - o2::pid::tof::ExpTimes<Trks::iterator, PID::Pion, false>::GetExpectedSignal(t));
+      histos.fill(HIST("goodforevtime/expP"), t.p(), t.tofExpMom());
+      histos.fill(HIST("goodforevtime/mass"), mass);
+      histos.fill(HIST("goodforevtime/tofSignalPerCollision"), ncolls % 6000, t.tofSignal());
+    }
+  }
+};
+
+WorkflowSpec defineDataProcessing(ConfigContext const& cfgc)
+{
+  auto workflow = WorkflowSpec{adaptAnalysisTask<tofSignal<false>>(cfgc),
```

Comment:
Is the only difference between the Run 2 and Run 3 workflow the `true` or `false`?
In the future the run 2 and run 3 should be decided by the AOD metadata and therefore it would be better if it is in the same file and just configures itself based on a PROCESS_SWITCH for now...